### PR TITLE
add scheduling constraint for kube nodes

### PR DESCRIPTION
**Pull request by corindwyer**
### Description of the Change
Add scheduling constraint for kube nodes instead of trying to do lease management when conditions and taints change.

### CODE REVIEW

File: titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java [hunk 1]
```diff
@@ -329,18 +329,10 @@ public void onDelete(V1Node node, boolean deletedFinalStateUnknown) {
 
     private void nodeUpdated(V1Node node) {
         try {
-            boolean removeStopped = node.getStatus().getConditions().stream()
```
Comment (tbak) [inline titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java:332]: Do we depend on GC process here to do the cleanup?

File: titus-server-master/src/main/java/com/netflix/titus/master/scheduler/constraint/KubeConstraint.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            return VALID;
+        }
+
+        Optional<AgentInstance> instanceOpt = SchedulerUtils.findInstance(agentManagementService, schedulerConfiguration.getInstanceAttributeName(), targetVM);
+        if (!instanceOpt.isPresent()) {
+            return Failure.INSTANCE_NOT_FOUND.toResult();
+        }
+
+        String instanceId = instanceOpt.get().getId();
+        V1Node node = kubeApiFacade.getNodeInformer().getIndexer().getByKey(instanceId);
```
Comment (tbak) [inline titus-server-master/src/main/java/com/netflix/titus/master/scheduler/constraint/KubeConstraint.java:116]: What if node is not there? We should check for null at least + double check that an exception is not thrown if a node is missing or due to other problems.

File: titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java (hunk 1 above)
Comment (corindwyer) [reply titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeApiServerIntegrator.java:332]: Yes.

File: titus-server-master/src/main/java/com/netflix/titus/master/scheduler/constraint/KubeConstraint.java (hunk 2 above)
Comment (corindwyer) [reply titus-server-master/src/main/java/com/netflix/titus/master/scheduler/constraint/KubeConstraint.java:118]: good catch. updated

**Pull request merged**